app/javascript/controllers/spinner_controller.js

```diff
@@ -0,0 +1,28 @@
+import { Controller } from "@hotwired/stimulus";
+
+function preventEscapeListener(event) {
+  if (event.key === "Escape") {
+    event.preventDefault();
+    event.stopPropagation();
+  }
+}
+
+export default class extends Controller {
+  static targets = ['submit'];
+
+  submitStart() {
+    document.addEventListener("keydown", preventEscapeListener, true);
+    document.querySelector(".dialog--header").classList.add("opacity-20");
+    document.querySelector(".dialog--section > *:not(:last-child)").classList.add("opacity-20");
+    document.querySelector(".dialog--close").classList.add("hidden");
+    document.querySelector("#spinner").classList.remove("hidden");
+  }
+
+  submitEnd() {
+    document.addEventListener("turbo:submit-end", preventEscapeListener);
```

Comment:
**question:** Should this be `removeEventListener` for `keydown`?

Reply:
ooops.... 4070dbe4962692db1487a6122176257a5d37926b